cg/apps/demultiplex/sample_sheet/sample_sheet_validator.py

```diff
@@ -0,0 +1,172 @@
+            LOG.error(message)
+            raise SampleSheetError(message)
+
+    def _get_index_settings_name(self) -> str:
+        """
+        Find the entry in the sample sheet holding the index settings name, which has the form:
+        `IndexSettings,<index_setting_name>`, and extract its value.
+        """
+
+        for row in self.content:
+            if SampleSheetBCLConvertSections.Header.INDEX_SETTINGS in row:
+                return row[1]
+        message: str = "No index settings found in sample sheet"
+        LOG.error(message)
+        raise SampleSheetError(message)
+
+    def _set_is_index2_reverse_complement(self) -> None:
+        """Return whether the index2 override cycles value is reverse-complemented."""
+        settings_name: str = self._get_index_settings_name()
+        self.is_index2_reverse_complement = NAME_TO_INDEX_SETTINGS[
+            settings_name
+        ].are_i5_override_cycles_reverse_complemented
+
+    def _get_cycle(self, cycle_name: str, nullable: bool = False) -> int | None:
+        """
+        Return the cycle from the sample sheet given the cycle name. Set nullable to True to
+        return None if the cycle is not found.
+        Raises:
+            SampleSheetError if the cycle is not found and nullable is False.
+        """
+        for row in self.content:
+            if cycle_name in row:
+                return int(row[1])
+        if not nullable:
+            message: str = f"No {cycle_name} found in sample sheet"
+            LOG.error(message)
+            raise SampleSheetError(message)
+
+    def _set_cycles(self):
+        """Set values to the run cycle attributes."""
+        self.read1_cycles = self._get_cycle(SampleSheetBCLConvertSections.Reads.READ_CYCLES_1)
+        self.read2_cycles = self._get_cycle(SampleSheetBCLConvertSections.Reads.READ_CYCLES_2)
+        self.index1_cycles = self._get_cycle(SampleSheetBCLConvertSections.Reads.INDEX_CYCLES_1)
+        self.index2_cycles = self._get_cycle(
+            cycle_name=SampleSheetBCLConvertSections.Reads.INDEX_CYCLES_2, nullable=True
+        )
+
+    def _validate_samples(self) -> None:
+        """
+        Determine if the samples have the correct attributes and are not unique per lane.
+            Raises:
+            ValidationError if the samples do not have the correct attributes based on their model.
+            SampleSheetError if the samples are not unique per lane.
+        """
+        validated_samples: list[FlowCellSample] = get_flow_cell_samples_from_content(self.content)
+        validate_samples_unique_per_lane(validated_samples)
+
+    def _validate_override_cycles(self) -> None:
+        """Determine if the samples' override cycles are valid.
+        Raises:
+            SampleSheetError if any of the samples' override cycles are not valid.
+        """
+        samples: list[dict[str, str]] = get_raw_samples(self.content)
+        validator = OverrideCyclesValidator(
```

Comment:
Design issue: tight coupling. Pass via constructor to `SampleSheetValidator`

Reply:
Big refactoring, left for another PR of unification of sample sheet models